static/js/main.js

```diff
@@ -0,0 +1,253 @@
+            displayLoader(false);
+        });
+}
+
+const displayMoreCountries = () => {
+    if (!maxPagesReached && !loadingData && Math.abs(document.documentElement.scrollHeight - document.documentElement.scrollTop - document.documentElement.clientHeight) <= 25) {
+        const startIndex = pageCount * cardsPerPage;
+        const endIndex = (pageCount * cardsPerPage) + cardsPerPage;
+        let maxPages;
+        pageCount++;
+        displayLoader(true);
+
+        fetchData(cardInfo)
+            .then((data) => {
+                let filteredData = data;
+
+                if (filterActive && !searchActive) {
+                    filteredData = data.filter((country) => country.region.toLowerCase() === activeFilter);
+                } else if (searchActive && !filterActive) {
+                    filteredData = data.filter((country) => country.name.toLowerCase().includes(activeSearch.toLowerCase()));
+                }
+
+                maxPages = Math.ceil(filteredData.length / cardsPerPage);
+
+                if (pageCount <= maxPages) {
+                    const moreCountryData = filteredData.slice(startIndex, endIndex);
+                    displayCards(moreCountryData);
+                } else {
+                    maxPagesReached = true;
+                }
+
+                displayLoader(false);
+            });
+
+    }
+}
+
+const displayCards = (countries) => {
+    const cards = [];
+
+    countries.forEach((country) => {
+
+        let countryCapital = country.capital;
+
+        if (countryCapital === undefined) {
+            countryCapital = 'None';
+        }
+
+        const card = document.createElement('div');
+        card.classList.add('country-card');
+        card.innerHTML = ` 
+        <img class="country-flag" src="${country.flag}">
+        <h2 class="country-name">${country.name}</h2>
+        <ul class="card-ul">
+            <li class="card-li">
+                <span class="span-category">Population:</span>
+                <span class="span-value">${country.population.toLocaleString('en-US')}</span>
+            </li>
+            <li class="card-li">
+                <span class="span-category">Region:</span>
+                <span class="span-value">${country.region}</span>
+            </li>
+            <li class="card-li">
+                <span class="span-category">Capital:</span>
+                <span class="span-value">${countryCapital}</span>
+            </li>
+        </ul>
+        `;
+
+        cards.push(card);
+    });
+
+    cards.forEach((card) => {
+        countryCardContainer.appendChild(card);
+    });
+}
+
+const addRegionsToSelectDropdown = () => {
+    const regions = [];
+    fetchData(['region'])
+        .then((data => {
+            data.forEach((country) => {
+                if (!regions.includes(country.region)) {
+                    regions.push(country.region);
+                }
+            });
+
+            regions.forEach((region) => {
```

Comment:
Ik denk dat je hiervoor ook de "template literals html" kan gebruiken. in plaats van creating elements and setting attributes.